### DISCUSSION TITLE
Tree like relationships

### DISCUSSION THREAD

**Discussion by philhawthorn** [noam-honig/radweb#308] [Q&A] [answered]
Hi, I'm wondering what the best way to approach a recursive relationship entity.  E.g. like a folder/files tree like structure?

I have the following entity:

```
@Entity("folders", {
    allowApiCrud: true
})
export class Folder {
    @Fields.autoIncrement()
    id!: number;
    @Fields.string()
    name!: string;
    @Relations.toOne(()=> User, {
        defaultIncluded: true
    })
    owner?: User;
    @Relations.toOne(()=> Folder, {
        defaultIncluded: true
    })
    parent?: Folder;
    @Relations.toMany<Folder, Folder>(()=>Folder, "parent")
    folders?: Folder[] 
}
```

And the client code, which of course fetches the first level (and I am aware that I can include a pre-determined depth):

```
const folders = await remult.repo(Folder).find({
  include: {
    folders: true
  }
})

```

Without remult, I would have queried all records and then organised them hierarchically - I'm wondering if there's an elegant way of handling in remult, if if I would have to do the same? 

Many thanks

**Comment by noam-honig** [comment, marked as answer]
You can set the `defaultIncluded` of the `toMany` relations to true and works nice - see:
https://codesandbox.io/p/devbox/demo-tree-shape-qvxkzv
(I had to set the defaultIncluded of the toOne relation to false for this

**Reply by philhawthorn** [reply]
Ah, thanks so much - I swear I already tried that, but perhaps not. The library is awesome, very slick 👏

**Reply by noam-honig** [reply]
Thanks - while I was working on it, I ran into a bug with default included - I'm working on it now and will release a version later today

**Reply by noam-honig** [reply]
Just released version 0.23.25 that will work even if you do keep the defaultIncluded for the `toOne` relation, and will also filter well on null - even if do not specify allowNull true

**Reply by philhawthorn** [reply]
Wow, amazing - thank you so much, thats really helpful.

**Comment by SpeedoPasanen** [comment]
If you eagerly load parent, does also parent.parent etc get loaded recursively ie. potentially a lot of queries?

**Comment by noam-honig** [comment]
Yes, if that becomes an issue you can selectively include - by setting
default included false
